### PR TITLE
Add a basic microstates type

### PR DESCRIPTION
**Pull request by taras**
This is the beginning of better TypeScript support with Microstates. In this PR, I added a microstates.d.ts file that covers a few exports from the microstates package. It's not much but it's something.

### CODE REVIEW

File: types/microstates.d.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    state: boolean
+  }
+
+  export interface StringType {
+    concat(): StringType
+    set(value: string): StringType
+    state: string
+  }
+
+  export function create<T>(Type: Constructor<T>, value?: any): T
```
Comment (cowboyd) [inline types/microstates.d.ts]: Shouldn't the return type be `T & Any`

File: types/microstates.d.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  export function Store<T>(microstate: T, next: (state: T) => any): T
```
Comment (cowboyd) [inline types/microstates.d.ts]: Again, should be `T & Any`?

File: types/microstates.d.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    new (...args): T;
+  }
+  
+  export interface Any {
+    set(value): Any
+  }
+  
+  export interface BooleanType {
+    increment(): BooleanType
+    decrement(): BooleanType
```
Comment (cowboyd) [inline types/microstates.d.ts]: ?

Comment (taras) [reply types/microstates.d.ts]: oops

File: types/microstates.d.ts (hunk 2 above)
Comment (taras) [reply types/microstates.d.ts]: Yep, corrected.

**Comment by taras** [comment]
@cowboyd I think this might only be the case for chaining transitions in custom types. I'm not sure.

**Pull request merged**